agenta-backend/agenta_backend/tasks/evaluations.py

```diff
@@ -312,45 +313,15 @@ async def aggregate_evaluator_results(
 
         if not results:
             result = Result(type="error", value=None, error=Error(message="-"))
-        else:
-            if evaluator_key == "auto_ai_critique":
-                numeric_scores = []
-                for result in results:
-                    # Extract the first number found in the result value
-                    match = re.search(r"\d+", result.value)
-                    if match:
-                        try:
-                            score = int(match.group())
-                            numeric_scores.append(score)
-                        except ValueError:
-                            # Ignore if the extracted value is not an integer
-                            continue
-
-                # Calculate the average of numeric scores if any are present
-                average_value = (
-                    sum(numeric_scores) / len(numeric_scores)
-                    if numeric_scores
-                    else None
-                )
-                result = Result(
-                    type="number",
-                    value=average_value,
-                )
 
-            else:
-                # Handle boolean values for auto_regex_test and other evaluators
-                if all(isinstance(result.value, bool) for result in results):
-                    average_value = sum(result.value for result in results) / len(
-                        results
-                    )
-                else:
-                    # Handle other data types or mixed results
-                    average_value = None
+        if evaluator_key == "auto_ai_critique":
+            result = aggregation_service.aggregate_ai_critique(results)
 
-                result = Result(
-                    type="number",
-                    value=average_value,
-                )
+        if evaluator_key == "auto_regex_test":
+            result = aggregation_service.aggregate_auto_regex(results)
+
+        if evaluator_key not in ["auto_ai_critique", "auto_regex_test"]:
+            result = aggregation_service.aggregate_evaluator(results)
```

Comment:
else if it is not in any of the evaluators please raise an exception, this makes sure that we don't have a bug later when we add a new evaluator and per default we use this one.